avaframe/com1DFAPy/com1DFA.py

```diff
@@ -1707,6 +1715,58 @@ def readPartFromPickle(inDir, flagAvaDir=False):
     return Particles, TimeStepInfo
 
 
+def savePartToCsv(particleProperties, dictList, outDir):
+    """ Save each particle dictionary from a list to a csv file; works also for one dictionary instead of list
+
+        Parameters
+        ---------
+        dictList: list or dict
+            list of dictionaries or single dictionary
+    """
+
+    # set output directory
+    outDir = os.path.join(outDir, 'particlesCSV')
+    fU.makeADir(outDir)
+
+    # read particle properties to be saved
+    particleProperties = particleProperties.split('|')
+
+    # write particles locations and properties to csv file
+    nParticles = len(dictList)
+    count = 0
+    for m in range(nParticles):
+        particles = dictList[count]
+        x = particles['x'] + particles['xllcenter']
+        y = particles['y'] + particles['yllcenter']
+        z = particles['z']
+        simName = particles['simName']
+        partPropDict = {}
+
+        for partProp in particleProperties:
+            if partProp == 'velocityMagnitude':
+                ux = particles['ux']
+                uy = particles['uy']
+                uz = particles['uz']
+                partPropDict[partProp] = DFAtls.norm(ux, uy, uz)
+            else:
+                partPropDict[partProp] = particles[partProp]
+
+
+        with open(os.path.join(outDir, 'particles%s.csv.%d' % (simName, count)), 'w') as pFile:
+                pFile.write('X, Y, Z,')
+                for m in range(len(particleProperties)-1):
+                    pFile.write('%s,' % (particleProperties[m]))
+                pFile.write('%s\n' % (particleProperties[-1]))
+
+                for k in range(len(x)):
+                    pFile.write('%.8f, %.8f, %.8f,' % (x[k], y[k], z[k]))
+                    for m in range(len(particleProperties)-1):
+                        pFile.write('%.8f,' % (partPropDict[particleProperties[m]][k]))
+                    pFile.write('%.8f\n' % (partPropDict[particleProperties[-1]][k]))
+
+                count = count + 1
```

Comment:
This seems to be a bit too complicated. I suggest splitting it up and/or use an existing function like eg. `to_csv` from pandas, the standard csv.write (https://docs.python.org/3/library/csv.html) or similar.

Reply:
done